shenyu-plugin/shenyu-plugin-base/src/main/java/org/apache/shenyu/plugin/base/cache/MetaDataCache.java

```diff
@@ -70,11 +70,16 @@ public static MetaDataCache getInstance() {
      * @param data the data
      */
     public void cache(final MetaData data) {
-        META_DATA_MAP.put(data.getPath(), data);
-        // the update is also need to clean, but there is
-        // no way to distinguish between crate and update,
-        // so it is always clean
+        // clean old path data
+        if (META_DATA_MAP.containsKey(data.getId())) {
+            // the update is also need to clean, but there is
+            // no way to distinguish between crate and update,
+            // so it is always clean
+            clean(META_DATA_MAP.get(data.getId()).getPath());
+        }
+        META_DATA_MAP.put(data.getId(), data);
         clean(data.getPath());
+        clean(DIVIDE_CACHE_KEY);
```

Comment:
In `DividePluginDataHandler`, we had call `MetaDataCache#clean()`, Can `clean(DIVIDE_CACHE_KEY);` be deleted here?